### ISSUE TITLE
Ctrl + C can not terminate the program 

### ISSUE THREAD

**Issue by 15663671003** [confluentinc/confluent-kafka-python#209]
Ctrl + C can not terminate the program.
The program may block the signal set inside
may be blocked in the consumer.poll () call, this method to solve

**Comment by DXist** [comment]
+1

**Comment by 15663671003** [comment]
Set consumer.poll (timeout = 1) can be resolved

**Comment by DXist** [comment]
Yes, it solves the problem. This issue is more about improving the documentation.

**Comment by edenhill** [comment]
There is currently no way to cancel a blocking call into librdkafka, when this is added to librdkafka we'll make the proper changes to the Python module to fix it.

**Comment by fpietka** [comment]
@edenhill is there by any chance any related issue in edenhill/librdkafka ?

**Comment by edenhill** [comment]
The closest thing would be https://github.com/edenhill/librdkafka/issues/614

**Comment by royreznik** [comment]
Is there anythin that going to be done about this?

**Comment by edenhill** [comment]
Yes, but it is quite involved and requires changes to librdkafka.
